### PR TITLE
Comms Vat 

### PR DESCRIPTION
**Pull request by katelynsills**
This PR is to put the commsVat changes on trunk quickly. There are lot of changes in this PR due to the rapid iteration, so this is not ideal. (Let me know if you want me to split this up into multiple PRs.)

These changes include:

* switching to "ingresses" and "egresses"
* using two tables, not four
* using a three-way mapping in clists with much clearer names
* handling promises and resolvers as slots

It doesn't:
* handle the case of an argument slot with a promise for presence (that test is skipped)
* necessarily do the right thing when it comes to POLA choices for the structure of the clist. For instance, there really aren't 'ingress' and 'egress' tables, per se, just types in the incoming/outgoing wire message ('your-ingress', 'your-egress') that are mapped correctly. This may be ok but probably isn't and deserves another look now the functionality seems to be working and it's not as confusing anymore.

### CODE REVIEW

File: src/kernel/kernel.js [hunk 1]
```diff
@@ -86,7 +86,7 @@ export default function buildKernel(kernelEndowments) {
       } else if (kp.state === 'fulfilledToData') {
         const s = `data is not callable, has no method ${msg.method}`;
         // eslint-disable-next-line no-use-before-define
-        reject(msg.kernelPromiseID, makeError(s), []);
+        reject(target.id, makeError(s), []);
```
Comment (warner) [inline src/kernel/kernel.js]: can you tell me about this change? I don't know that we have a specific test for it, but the intention was that sending a message to a promise that was previously fulfilled to data (obviously an error) causes the sender's promise to get rejected. I think `target.id` is the target that's been resolved-to-data, so I don't see how we can reject it after it's already been resolved. I think it should be doing roughly the same as the `kp.state === 'rejected'` case below, except with a static error.

Comment (katelynsills) [reply src/kernel/kernel.js]: Ah, good point, I think it was necessary at some time in the past, but that might have been with a mistaken assumption. I reverted the change.

File: docs/comms-vat.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Anything object that is on our machine that we want to allow another
+machine to reference is an EGRESS. Likewise, any object that we want to
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply docs/comms-vat.md]: I think we may be talking about different things - tables verses types. I changed it to explicitly refer to the ingress or egress types that appear in over the wire messages, and made some additional changes to try to incorporate the language you used here. Does that look ok?

File: docs/comms-vat.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+reference on another machine is an INGRESS. Ingresses and egresses can
+appear in the slots of messages going either way - in other words, the
+arguments of my message can be referencing either things on my machine
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply docs/comms-vat.md]: Yup! Added that in

File: docs/comms-vat.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+or your machine. 
+
+In the same way that Egresses are real objects that have agency, and Ingresses are
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply docs/comms-vat.md]: Yeah, that wording doesn't make much sense, I've taken it out

File: docs/comms-vat.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+presences, we can send messages about Promises and Resolvers.
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply docs/comms-vat.md]: I think this is also a tables vs types issue

File: src/devices.js [hunk 6]
```diff
@@ -81,7 +81,7 @@ export function buildChannel() {
         try {
           callback(`${fromMachineName}`, `${data}`);
         } catch (err) {
-          console.log(`error during callback: ${e} ${e.message}`);
+          console.log(`error during callback: ${err} ${err.stack}`, err);
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply src/devices.js]: Good point, I'll revert that!

**Pull request merged**